src/test/regress/expected/multi_subquery_window_functions.out

```diff
@@ -0,0 +1,745 @@
+    *, rank() OVER my_win as rnk
+  FROM
+    events_table
+    WINDOW my_win AS (ORDER BY time DESC)
+) as foo
+ORDER BY
+  3 DESC, 1 DESC, 2 DESC
+LIMIT
+  10;
+ERROR:  cannot push down this subquery
+DETAIL:  Window Functions without partition by on distribution column is currently unsupported
+-- w2 should not be pushed down
+SELECT * FROM
+(
+  SELECT
+    DISTINCT users_table.user_id, lag(users_table.user_id) OVER w1, rank() OVER w2
+  FROM
+    users_table, events_table
+  WHERE
+    users_table.user_id = events_table.user_id and
+    event_type < 25
+  WINDOW w1 AS (PARTITION BY users_table.user_id, events_table.event_type ORDER BY events_table.time),
+  w2 AS (PARTITION BY users_table.user_id+1, (events_table.value_2 % 25) ORDER BY events_table.time)
+) as foo
+ORDER BY 3 DESC, 1 DESC, 2 DESC NULLS LAST
+LIMIT 10;
+ERROR:  cannot push down this subquery
+DETAIL:  Partition by list without partition column is currently unsupported
+-- w2 should not be pushed down
+SELECT * FROM
+(
+  SELECT
+    DISTINCT users_table.user_id, lag(users_table.user_id) OVER w1, rank() OVER w2
+  FROM
+    users_table, events_table
+  WHERE
+    users_table.user_id = events_table.user_id and
+    event_type < 25
+  WINDOW w1 AS (PARTITION BY users_table.user_id, events_table.event_type ORDER BY events_table.time),
+  w2 AS (ORDER BY events_table.time)
+) as foo
+ORDER BY
+  3 DESC, 1 DESC, 2 DESC NULLS LAST
+LIMIT
+  10;
+ERROR:  cannot push down this subquery
+DETAIL:  Window Functions without partition by on distribution column is currently unsupported
+-- GROUP BY includes the partition key, but not the WINDOW function
+SELECT
+   user_id, time, my_rank
+FROM
+(
+  SELECT
+    user_id, date_trunc('day', time) as time, rank() OVER my_win as my_rank
+  FROM
+    events_table
+  GROUP BY
+    user_id, date_trunc('day', time)
+    WINDOW my_win AS (ORDER BY avg(event_type))
+) as foo
+WHERE
+  my_rank > 125
+ORDER BY
+  3 DESC, 1 DESC,2 DESC
+LIMIT
+  10;
+ERROR:  cannot push down this subquery
+DETAIL:  Window Functions without partition by on distribution column is currently unsupported
+-- GROUP BY includes the partition key, but not the WINDOW function
+SELECT
+   user_id, time, my_rank
+FROM
+(
+  SELECT
+    user_id, date_trunc('day', time) as time, rank() OVER my_win as my_rank
+  FROM
+    events_table
+  GROUP BY
+    user_id, date_trunc('day', time)
+    WINDOW my_win AS (PARTITION BY date_trunc('day', time) ORDER BY avg(event_type))
+) as foo
+WHERE
+  my_rank > 125
+ORDER BY
+  3 DESC, 1 DESC,2 DESC
+LIMIT
+  10;
+ERROR:  cannot push down this subquery
+DETAIL:  Partition by list without partition column is currently unsupported
+-- Using previously defined supported window function on distribution key
+SELECT * FROM (
+  SELECT
+    user_id, date_trunc('day', time) as time, sum(rank) OVER w2
+  FROM (
+    SELECT DISTINCT
+      user_id as user_id, time, rank() over w1
+    FROM users_table
+    WINDOW
+      w AS (PARTITION BY user_id),
```

Comment:
Could you briefly explain the semantics of this usage where you define a window function using another window function. 

For instance, what would be the behavior if there were `ORDER BY` in `w`, or `w1` also includes `user_id` etc? 

 (In general, is the overriding used for changing the `ORDER BY`s or can we also change the `PARTITION BY`)?

Reply:
Citing from [Postgres documentation](https://www.postgresql.org/docs/10/static/sql-select.html)
> If an existing_window_name is specified it must refer to an earlier entry in the WINDOW list; the new window copies its partitioning clause from that entry, as well as its ordering clause if any. In this case the new window cannot specify its own PARTITION BY clause, and it can specify ORDER BY only if the copied window does not have one. The new window always uses its own frame clause; the copied window must not specify a frame clause.

So, if an earlier window function is referenced to create a new one, user is not able to modify partition by or order by clauses if they are already defined.